go/store/types/edits/tuple_stream_merger.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package edits
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+type entry struct {
+	kvp       *types.KVP
+	readerIdx int
+}
+
+var _ types.EditProvider = (*TupleStreamMerger)(nil)
+
+type TupleStreamMerger struct {
+	ctx       context.Context
+	nbf       *types.NomsBinFormat
+	numEdits  int64
+	editsRead int64
+
+	numReaders      int
+	readersWithData int
+	readers         []types.TupleReadCloser
+	nextKVPS        []entry
+
+	readerClosed func(i int)
+}
+
+// EditProviderForFiles takes a list of filenames whose contents are alternating key value tuples written out using a
+// TupleWriter and returns a TupleStreamMerger.  The tuples within each file must be sorted by the key
+// tuples in order for valid.  The streams of tuples are merge sorted as they are read.
```

Comment:
```suggestion
// tuples in order to be valid.  The streams of tuples are merge sorted as they are read.
```

Reply:
A double space in front of "The streams"

Reply:
THIS IS AGEISM:

https://www.cultofpedagogy.com/two-spaces-after-period/